### PR TITLE
ci: push podfile updates on renovate PRs

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

Many renovate PRs fail only on Podfile.lock diffs and require manual intervention to succeed. This updates the github action to automatically push the Podfile.lock changes for renovate PRs so they can automatically merge.

### Test plan

Manually on CI by changing the branch and username requirements to match this PR and pushing a renovate PR diff
- renovate PR commit - https://github.com/valora-inc/wallet/pull/4285/commits/762931c9ea01038fe2d10b8c105650f2e7472c23
- github action run pushing the Podfile.lock changes - https://github.com/valora-inc/wallet/actions/runs/6463824413/job/17547540940
- commit with Podfile changes - https://github.com/valora-inc/wallet/pull/4285/commits/5dab532af36617f7dba6172970a83c78cb1a1168 (action run passing the podfile.lock step https://github.com/valora-inc/wallet/actions/runs/6463850277/job/17547603233#step:12:18)

### Related issues

N/A

### Backwards compatibility

Yes

### CODE REVIEW

File: .github/workflows/e2e-ios.yml [hunk 1]
```diff
@@ -29,6 +29,10 @@ jobs:
           secrets: |-
             EMERGE_API_TOKEN:projects/1027349420744/secrets/EMERGE_API_TOKEN
       - uses: actions/checkout@v3
+        with:
+          # checkout PR head instead of merge commit so we can push changes on renovate PRs
+          # https://github.com/actions/checkout/tree/v3.6.0#checkout-pull-request-head-commit-instead-of-merge-commit
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (jeanregisser) [inline .github/workflows/e2e-ios.yml:35]: Should we make this conditional base on the renovatebot user then?

**Review by cajubelt — APPROVED** [review]

**Comment by cajubelt** [comment]
whoops, mistook this for a different PR

File: .github/workflows/e2e-ios.yml (hunk 1 above)
Comment (satish-ravi) [reply .github/workflows/e2e-ios.yml]: I ended up doing the latter, although it got a bit messy. lmk what you think

**Review by jeanregisser — APPROVED** [review]
w00t! This is great! 🚀 ❤️

**Pull request merged**